administrator/components/com_finder/src/Indexer/Indexer.php

```diff
@@ -977,13 +985,19 @@ protected function toggleTables($memory)
 
         // Check if we are setting the tables to the Memory engine.
         if ($memory === true && $state !== true) {
-            // Set the tokens table to Memory.
-            $db->setQuery('ALTER TABLE ' . $db->quoteName('#__finder_tokens') . ' ENGINE = MEMORY');
-            $db->execute();
+            try {
+                // Set the tokens table to Memory.
+                $db->setQuery('ALTER TABLE ' . $db->quoteName('#__finder_tokens') . ' ENGINE = MEMORY');
+                $db->execute();
 
-            // Set the tokens aggregate table to Memory.
-            $db->setQuery('ALTER TABLE ' . $db->quoteName('#__finder_tokens_aggregate') . ' ENGINE = MEMORY');
-            $db->execute();
+                // Set the tokens aggregate table to Memory.
+                $db->setQuery('ALTER TABLE ' . $db->quoteName('#__finder_tokens_aggregate') . ' ENGINE = MEMORY');
+                $db->execute();
+            } catch (\Joomla\Database\Exception\ExecutionFailureException $e) {
```

Comment:
```suggestion
            } catch (\RuntimeException $e) {
```
With the PDO driver the PR was working fine, but not with the MySQLi driver. The reason is that the MySQLi driver raised a `mysqli_sql_exception`. I have checked that with some debug output and a `get_class($e)` call.

As both the Joomla db ExecutionFailureException exception and the mysqli_sql_exception are derived from RuntimeException it works with that with both drivers.

The behaviour of the different drivers may also depend on database client versions.

It's similar to the issues I had once fixed with #39391 and #35951 .

Reply:
Okay, makes sense